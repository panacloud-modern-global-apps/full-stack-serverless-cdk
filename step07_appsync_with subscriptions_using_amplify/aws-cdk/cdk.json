{
  "app": "npx ts-node bin/aws-cdk.ts",
  "context": {
    "@aws-cdk/core:enableStackNameDuplicates": "true",
    "aws-cdk:enableDiffNoFail": "true",
    "@aws-cdk/core:stackRelativeExports": "true",
    "@aws-cdk/aws-ecr-assets:dockerIgnoreSupport": true
  }
}
