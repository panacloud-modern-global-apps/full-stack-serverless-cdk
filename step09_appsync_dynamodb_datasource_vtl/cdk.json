{
  "app": "npx ts-node bin/stepxx_appsync_dynamodb_datasource_vtl.ts",
  "context": {
    "@aws-cdk/core:enableStackNameDuplicates": "true",
    "aws-cdk:enableDiffNoFail": "true",
    "@aws-cdk/core:stackRelativeExports": "true",
    "@aws-cdk/aws-ecr-assets:dockerIgnoreSupport": true
  }
}
