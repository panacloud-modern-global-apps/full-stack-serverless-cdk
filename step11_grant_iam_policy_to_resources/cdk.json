{
  "app": "npx ts-node bin/stepxx_grant_iam_policy_to_resources.ts",
  "context": {
    "@aws-cdk/core:enableStackNameDuplicates": "true",
    "aws-cdk:enableDiffNoFail": "true",
    "@aws-cdk/core:stackRelativeExports": "true",
    "@aws-cdk/aws-ecr-assets:dockerIgnoreSupport": true
  }
}
