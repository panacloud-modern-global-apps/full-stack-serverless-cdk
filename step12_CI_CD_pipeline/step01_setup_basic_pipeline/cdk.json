{
  "app": "npx ts-node bin/step01_setup_basic_pipeline.ts",
  "context": {
    "@aws-cdk/core:enableStackNameDuplicates": "true",
    "aws-cdk:enableDiffNoFail": "true",
    "@aws-cdk/core:stackRelativeExports": "true",
    "@aws-cdk/aws-ecr-assets:dockerIgnoreSupport": true
  }
}
