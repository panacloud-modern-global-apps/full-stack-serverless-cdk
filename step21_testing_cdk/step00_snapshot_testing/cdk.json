{
  "app": "npx ts-node --prefer-ts-exts bin/step00_snapshot_testing.ts",
  "context": {
    "@aws-cdk/core:enableStackNameDuplicates": "true",
    "aws-cdk:enableDiffNoFail": "true",
    "@aws-cdk/core:stackRelativeExports": "true",
    "@aws-cdk/aws-ecr-assets:dockerIgnoreSupport": true
  }
}
