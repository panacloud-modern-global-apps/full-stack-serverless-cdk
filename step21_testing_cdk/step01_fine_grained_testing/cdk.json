{
  "app": "npx ts-node --prefer-ts-exts bin/step01_fine_grained_testing.ts",
  "context": {
    "@aws-cdk/core:enableStackNameDuplicates": "true",
    "aws-cdk:enableDiffNoFail": "true",
    "@aws-cdk/core:stackRelativeExports": "true",
    "@aws-cdk/aws-ecr-assets:dockerIgnoreSupport": true
  }
}
