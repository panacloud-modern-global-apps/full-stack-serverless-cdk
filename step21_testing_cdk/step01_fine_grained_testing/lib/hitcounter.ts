import * as cdk from '@aws-cdk/core';
import * as lambda from '@aws-cdk/aws-lambda';
import * as dynamodb from '@aws-cdk/aws-dynamodb';

export interface HitCounterProps {
    /** the function for which we want to count url hits **/
    downstream: lambda.IFunction;
}

export class HitCounter extends cdk.Construct {
    public readonly handler: lambda.Function;

    constructor(scope: cdk.Construct, id: string, props: HitCounterProps) {
        super(scope, id);
        const table = new dynamodb.Table(this, 'Hits', {
            partitionKey: { name: 'path', type: dynamodb.AttributeType.STRING }
        });

        this.handler = new lambda.Function(this, 'HitCounterHandler', {
            runtime: lambda.Runtime.NODEJS_10_X,
            handler: 'hitcounter.handler',
            code: lambda.Code.fromAsset('lambda'),
            environment: {
                DOWNSTREAM_FUNCTION_NAME: props.downstream.functionName,
                HITS_TABLE_NAME: table.tableName
            }
        });
        table.grantReadWriteData(this.handler);
        props.downstream.grantInvoke(this.handler);

    }
}
