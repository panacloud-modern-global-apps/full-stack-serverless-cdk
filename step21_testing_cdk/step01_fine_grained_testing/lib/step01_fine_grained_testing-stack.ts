import * as cdk from '@aws-cdk/core';
import * as lambda from '@aws-cdk/aws-lambda';
import * as apigw from '@aws-cdk/aws-apigateway';
import { HitCounter } from './hitcounter';

export class Step01FineGrainedTestingStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // defines an AWS Lambda resource
    const hello = new lambda.Function(this, 'HelloHandler', {
      runtime: lambda.Runtime.NODEJS_10_X,    // execution environment
      code: lambda.Code.fromAsset('lambda'),  // code loaded from "lambda" directory
      handler: 'hello.handler'                // file is "hello", function is "handler"
    });
    const helloWithCounter = new HitCounter(this, 'HelloHitCounter', {
      downstream: hello
    });
    // defines an API Gateway REST API resource backed by our "hello" function.
    new apigw.LambdaRestApi(this, 'Endpoint', {
      handler: helloWithCounter.handler
    });

  }
}
