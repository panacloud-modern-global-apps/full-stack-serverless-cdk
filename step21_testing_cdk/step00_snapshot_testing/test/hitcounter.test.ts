import { expect as expectCDK, haveResource } from '@aws-cdk/assert';
import cdk = require('@aws-cdk/core');
import * as lambda from '@aws-cdk/aws-lambda';
import { SynthUtils } from '@aws-cdk/assert';
import { HitCounter } from '../lib/hitcounter';

test('DynamoDB Table Created', () => {
    const stack = new cdk.Stack();
    // WHEN
    new HitCounter(stack, 'MyTestConstruct', {
        downstream: new lambda.Function(stack, 'TestFunction', {
            runtime: lambda.Runtime.NODEJS_10_X,
            handler: 'lambda.handler',
            code: lambda.Code.fromInline('test')
        })
    });
    // THEN
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();

});
