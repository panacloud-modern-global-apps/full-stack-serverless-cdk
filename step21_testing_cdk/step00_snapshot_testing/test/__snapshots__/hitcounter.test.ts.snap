// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamoDB Table Created 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6dArtifactHash4B0A8DC4": Object {
      "Description": "Artifact hash for asset \\"8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6d\\"",
      "Type": "String",
    },
    "AssetParameters8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6dS3Bucket70440D32": Object {
      "Description": "S3 bucket for asset \\"8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6d\\"",
      "Type": "String",
    },
    "AssetParameters8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6dS3VersionKeyE1929222": Object {
      "Description": "S3 key for asset version \\"8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6d\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "MyTestConstructHitCounterHandler17241061": Object {
      "DependsOn": Array [
        "MyTestConstructHitCounterHandlerServiceRoleDefaultPolicy8F2C3785",
        "MyTestConstructHitCounterHandlerServiceRoleD0F1215D",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6dS3Bucket70440D32",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6dS3VersionKeyE1929222",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters8579fdd339a0b4b233269e2a10fe0383b747bf73715b4565bc5c576e7d024c6dS3VersionKeyE1929222",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "DOWNSTREAM_FUNCTION_NAME": Object {
              "Ref": "TestFunction22AD90FC",
            },
            "HITS_TABLE_NAME": Object {
              "Ref": "MyTestConstructHits24A357F0",
            },
          },
        },
        "Handler": "hitcounter.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "MyTestConstructHitCounterHandlerServiceRoleD0F1215D",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "MyTestConstructHitCounterHandlerServiceRoleD0F1215D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MyTestConstructHitCounterHandlerServiceRoleDefaultPolicy8F2C3785": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "MyTestConstructHits24A357F0",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "TestFunction22AD90FC",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "MyTestConstructHitCounterHandlerServiceRoleDefaultPolicy8F2C3785",
        "Roles": Array [
          Object {
            "Ref": "MyTestConstructHitCounterHandlerServiceRoleD0F1215D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "MyTestConstructHits24A357F0": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "path",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "path",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "TestFunction22AD90FC": Object {
      "DependsOn": Array [
        "TestFunctionServiceRole6ABD93C7",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "test",
        },
        "Handler": "lambda.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "TestFunctionServiceRole6ABD93C7",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "TestFunctionServiceRole6ABD93C7": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
